ami.tf

```diff
@@ -5,32 +5,60 @@ locals {
     "AL2_x86_64" : "",
     "AL2_x86_64_GPU" : "-gpu",
     "AL2_ARM_64" : "-arm64",
+    "BOTTLEROCKET_x86_64" : "x86_64",
+    "BOTTLEROCKET_ARM_64" : "aarch64"
   }
 
+  ami_format = {
+    # amazon-eks{arch_label}-node-{ami_kubernetes_version}-v{ami_version}
+    # e.g. amazon-eks-arm64-node-1.21-v20211013
+    "AL2" : "amazon-eks%s-node-%s"
+    # bottlerocket-aws-k8s-{ami_kubernetes_version}-{arch_label}-v{ami_version}
+    # e.g. bottlerocket-aws-k8s-1.21-x86_64-v1.2.0-ccf1b754
+    "BOTTLEROCKET" : "bottlerocket-aws-k8s-%s-%s-%s"
+  }
+
+  ami_kind = split("_", var.ami_type)[0]
+
   # Kubernetes version priority (first one to be set wins)
   # 1. prefix of var.ami_release_version
   # 2. var.kubernetes_version
   # 3. data.eks_cluster.this.kubernetes_version
   need_cluster_kubernetes_version = local.enabled ? local.need_ami_id && length(concat(var.ami_release_version, var.kubernetes_version)) == 0 : false
 
-  ami_kubernetes_version = local.need_ami_id ? (local.need_cluster_kubernetes_version ? data.aws_eks_cluster.this[0].version :
-    regex("^(\\d+\\.\\d+)", coalesce(try(var.ami_release_version[0], null), try(var.kubernetes_version[0], null)))[0]
-  ) : ""
+  ami_kubernetes_version = local.need_ami_id ? {
+    "AL2" : (local.need_cluster_kubernetes_version ? data.aws_eks_cluster.this[0].version :
+      regex("^(\\d+\\.\\d+)", coalesce(try(var.ami_release_version[0], null), try(var.kubernetes_version[0], null)))[0]
+    ),
+    "BOTTLEROCKET" : var.kubernetes_version[0],
+  } : {}
```

Comment:
This needs improvement. As the comment says, we want to set the Kubernetes version based on the first of these sources that can provide it, in this order:
1. prefix of `var.ami_release_version`
2. `var.kubernetes_version`
3. `data.eks_cluster.this.kubernetes_version`

We can get the Kubernetes version from the `AL2` ami_release_version but not from the `BOTTLEROCKET` ami_release_version, which means we need to change how we compute `need_cluster_kubernetes_version`. Unfortunately, because that ends up being an input to `count`, we do not want to have it be dependent on the value of anything. So we have to be overly aggressive, and say
```hcl
need_cluster_kubernetes_version = local.enabled ? local.need_ami_id && length(var.kubernetes_version) == 0 : false
```

Now, though, we cannot use `need_cluster_kubernetes_version` to directly decide our priority. So we need an alternate variable like
```hcl
    use_cluster_kubernetes_version = local.need_cluster_kubernetes_version && (local.ami_kind == "BOTTLEROCKET" || length(var.ami_release_version) == 0)
```

then we need to fix the extraction
```hcl
  ami_kubernetes_version = local.need_ami_id ? (local.use_cluster_kubernetes_version ? data.aws_eks_cluster.this[0].version :
    regex("^(\\d+\\.\\d+)", coalesce(local.ami_kind == "AL2" ? try(var.ami_release_version[0], null) : null, try(var.kubernetes_version[0], null)))[0]
  ) : ""
```

Reply:
Thanks @Nuru. I addressed these changes and removed the reliance of `need_cluster_kubernetes_version` in `main.tf` by using `use_cluster_kubernetes_version` instead.